app/Http/Controllers/Rpc/WalletController.php

```diff
@@ -137,6 +140,38 @@ public function depositInfo()
         return self::json($resp);
     }
 
+    public function history()
+    {
+        $userId = Auth::user()->id;
+        $resp = [];
+        //TODO pagination
+        foreach (UserLedger::where('users_id', $userId)->cursor() as $ul) {
+            $amount = AdsConverter::clicksToAds($ul->amount);
+            $date = $ul->created_at->format(Carbon::RFC7231_FORMAT);
+            $txid = $ul->txid;
+            if (null !== $txid) {
+                // TODO export blockexplorer address / update testHistory in WalletControllerTest
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1135)